src/main/java/com/mg105/use_cases/BattleInteractor.java

```diff
@@ -0,0 +1,299 @@
+package com.mg105.use_cases;
+
+import com.mg105.entities.Battle;
+import com.mg105.entities.BattleCharacter;
+import com.mg105.entities.GameState;
+import com.mg105.entities.Move;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+/**
+ * This class interacts with the Battle entity.
+ * <p>
+ * It is responsible for modifying Battle related entities and for the processing needed during combat encounters.
+ */
+public class BattleInteractor {
+
+    private final GameState state;
+    private Battle encounter;
+
+    private BattlePresenterInterface presenter;
+
+    public BattleInteractor(GameState state) {
+        this.state = state;
+        this.encounter = state.getCurrEncounter();
+    }
+
+    /**
+     * Sets the presenter attribute.
+     *
+     * @param presenter the BattlePresenterInterface instance to set the presenter attribute to.
+     */
+    public void setPresenter(BattlePresenterInterface presenter) {
+        this.presenter = presenter;
+    }
+
+    /**
+     * Creates a new encounter with random opponents and sets it as the current encounter in GameState
+     */
+    public void createEncounter() {
+        Random rand = new Random();
+
+        ArrayList<BattleCharacter> opponents = new ArrayList<>();
+
+        for (int i = 0; i < 4; ++i){
+            int charHealth = rand.nextInt(5, 41);
+            int charDmg = rand.nextInt(1, 11);
+            int charSpeed = rand.nextInt(3, 16);
+            Move m1 = new Move(-rand.nextInt(1, 8), 0, "first", false);
+            Move m2 = new Move(rand.nextInt(1, 4), 0, "second", true);
+            BattleCharacter character = new BattleCharacter(charHealth, "Opponent " + i,
+                charDmg, charSpeed, true, m1, m2);
+            opponents.add(character);
+        }
+        ArrayList<BattleCharacter> party = this.state.getParty();
+        Battle b = new Battle(opponents, party);
+        this.state.setCurrEncounter(b);
+        this.encounter = b;
+
+        String[] partyNames = new String[party.size()];
+        String[] opponentNames = new String[opponents.size()];
+
+        for (int i = 0; i < party.size(); ++i) {
+            partyNames[i] = party.get(i).getName();
+        }
+
+        for (int i = 0; i < opponents.size(); ++i) {
+            opponentNames[i] = opponents.get(i).getName();
+        }
+
+        if (this.presenter == null) { //Used for unit tests. Never executed during regular running of the application.
```

Comment:
You actually don't need this for unit tests. You can sort of define an abstract of unimplmented interface (Look at inventory interactor test) but not really a big deal.

Reply:
Thank you.